massdriver.yaml

```diff
@@ -0,0 +1,69 @@
+schema: draft-07
+name: aws-sagemaker-inference-endpoint
+description: AWS SageMaker Inference Endpoint for hosting an AI model
+source_url: github.com/massdriver-cloud/aws-sagemaker-inference-endpoint
+access: public
+type: infrastructure
+
+params:
+  required:
+    - endpoint_config
+  properties:
+    endpoint_config:
+      title: Endpoint Configuration
+      type: object
+      required:
+        - primary_container
+        - instance_type
+        - instance_count
+      properties:
+        primary_container:
+          title: Primary Container
+          type: object
+          required:
+            - model_data
+            - ecr_image
+          properties:
+            model_data:
+              title: Model Data S3 URI
+              description: "The S3 URI of the model data. (e.g. s3://bucket-name/model.tar.gz)"
+              type: string
+            ecr_image:
+              title: ECR Image URI
+              description: "The ECR Image URI. (e.g. 	763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.1.0-gpu-py310-cu118-ubuntu20.04-ec2)"
+              type: string
+        instance_type:
+          title: Instance Type
+          description: "The instance type. (e.g. 	ml.g5.4xlarge)"
+          type: string
```

Comment:
Def want to turn this into a `oneOf` with the instance options, like this: https://github.com/massdriver-cloud/aws-eks-cluster/blob/main/massdriver.yaml#L149-L197

Reply:
Definitely can. But it's an extensive list and I'd expect will change a lot in the coming year.

https://aws.amazon.com/sagemaker/pricing/

Reply:
Have added an enumerated list that seems to cover a good variety of use cases.